src/wrap.tsx

```diff
@@ -22,123 +21,122 @@ afterEach(() => {
 })
 
 const wrap = (Component: typeof React.Component): Wrap => {
-  const options = {
-    Component,
+  updateOptions({
+    Component: Component,
     responses: [],
     props: {},
     path: '',
     hasPath: false,
     debug: process.env.npm_config_debugRequests === 'true',
-  }
+  })
 
-  return wrapWith(options)
+  return wrapWith()
 }
 
-const wrapWith = (options: WrapOptions): Wrap => {
-  const { extend, portal, changeRoute, history, mount } = getConfig()
-  const extensions = extendWith(extend, options)
+const wrapWith = (): Wrap => {
+  const extensions = extendWith()
 
   return {
-    withProps: getWithProps(options),
-    withNetwork: getWithNetwork(options),
-    atPath: getAtPath(options),
-    debugRequests: getDebugRequest(options),
-    mount: getMount(options, mount, changeRoute, history, portal),
+    withProps,
+    withNetwork,
+    atPath,
+    debugRequests,
+    mount: getMount,
     ...extensions,
   }
 }
 
-const addResponses = (options: WrapOptions) => (responses: Response[]) => {
-  options.responses = [...options.responses, ...responses]
+const addResponses = (newResponses: Response[]) => {
+  const options = getOptions()
+  const responses = [...options.responses, ...newResponses]
+
+  updateOptions({ ...options, responses })
```

Comment:
Why do you need to pass the options?
```suggestion
  const responses = [...options.responses, ...newResponses]

  updateOptions({ responses })
```

Reply:
As a first step, we create a function that receives an options model. Maybe in the future, we can have different methods to update the status of options like `updateResponses` to handle these cases.